Modules/_collectionsmodule.c

```diff
@@ -598,12 +609,16 @@ deque_copy_impl(dequeobject *deque)
         if (new_deque == NULL)
             return NULL;
         new_deque->maxlen = old_deque->maxlen;
-        /* Fast path for the deque_repeat() common case where len(deque) == 1 */
+        /* Fast path for the deque_repeat() common case where len(deque) == 1
+         *
+         * It's safe to not acquire the per-object lock for new_deque; it's
+         * invisible to other threads.
```

Comment:
Observation: comments like this makes me want to reconsider the naming previously discussed; future readers of the code will definitely scratch their heads and wonder if this comment is correct when they follow `deque_append_impl` and discover `deque_append_lock_held` 😉

(I'm not saying we _should_ reconsider the naming, but I think stuff like this is a good argument in favour of another naming scheme.)